### PR TITLE
Skip ssl certificate verif

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary
- add an option to skip SSL certificate verification when connecting to opensearch. This is useful when as a developer/tester we use the docker compose deployment which starts opensearch with ssl certificates which are only valid when requesting opensearch from the internal docker network (as elasticsearch:9200), which is not the case when one develop in an Eclipse IDE deployed on the host (e.g. developer's laptop).
- bump the version to 1.0.0-SNAPSHOT to get ready for v1
- 
## ⚙️ Test Data and/or Report
One of the following should be included here:
* Reference to regression test included in code (preferred wherever reasonable)
* Attach test data here + outputs of tests

### CODE REVIEW

**Review by al-niessner — CHANGES_REQUESTED** [review]
Is it possible to rebase along main to hide all of the opensearch changes? If not, then let me know and I will spend more time on sifting through the changes.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (tdddblog) [inline service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: Constructor has too many parameters. Consider using "builder" design pattern or pass a properties / config / context object.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (tdddblog) [inline service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java:99]: You only disable host name verification. Don't we also want to trust self-signed certificates?

**Review by tdddblog — CHANGES_REQUESTED** [review]

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: Actually, should get rid of this constructor altogether -- OpenSearchRegistryConnectionImpl(). The only user of it is TestLidVidUtils which can just hard code this solution.

The only user of other constructor is OpenSearchConfig. If the intent is to say that these values must be supplied and do not have valid or decent defaults (none of them are final which is the other reason to be in the constructor) then the builder pattern is not a good fit here. OpenSearchConfig, being the only caller of the constructor, supplies all values making the current solution sufficient.

Comment (tloubrieu-jpl) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: Actually I remember I create an interface/implementation design while I was struggling with the spring framework to manage the configuration. That might not be useful indeed.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (tloubrieu-jpl) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java:99]: @tdddblog I did just what I needed for the api to work with the opensearch within docker network. If the hostname trick works maybe we can just dismiss this pull request.

**Comment by tloubrieu-jpl** [comment]
> Is it possible to rebase along main to hide all of the opensearch changes? If not, then let me know and I will spend more time on sifting through the changes.

@al-niessner i made the rebase

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (tloubrieu-jpl) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java:99]: @tdddblog I changed the name of the property to be more explicit on the fact that we only disable the hostname verification.

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (tloubrieu-jpl) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: @tdddblog @al-niessner I implemented the builder design pattern

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

Sorry but the huge number of files that are changing really make me nervous in a bad way. Is this the only [changeset](https://github.com/NASA-PDS/registry-api/pull/110/commits/a7a5d32cb2a36b11ec363cd7f46f567fd6ccaad3) that is needed for the SSL changes and all others are merging and what not (you are not the committer in the rest)?

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

Actually it looks like these three commits are all that counts:

ssl

Is that correct? If you do not want to do it then I will (I am familiar with doing it), but it would be could to make a patch and start over. With the patch in hand, branch from current HEAD of main, apply patch, make new PR and lets review just the files that are changed. Something is not right with changing 52 files and I am concerned about regressing main to an desirable point in time.

**Review by al-niessner — CHANGES_REQUESTED** [review]
See comments about starting afresh

File: service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/opensearch/OpenSearchRegistryConnectionImpl.java]: I would recommend moving to the suggested Splitter class. The idiosyncrasies that this check identifies is exactly the kind that host will produce.

**Comment by tloubrieu-jpl** [comment]
@al-niessner sorry I thought I've done the rebase but it did not work. I did it again and the PR looks better, only showing the changes I actually made.

**Review by al-niessner — APPROVED** [review]

**Pull request merged**
